src/main/clojure/clara/benchmark/accum_repeatedly_not_selected.clj

```diff
@@ -0,0 +1,24 @@
+(ns clara.benchmark.accum-repeatedly-not-selected
+  "Added to test performance improvements in clara-rules issue 182"
+  (:require [clara.rules :refer :all]
+            [clara.rules.accumulators :as acc])
+  (:import [clara.benchmark.beans Discount Customer]
+           [java.util
+            ArrayList
+            List]))
+
+(def min-id (acc/min (fn [^Customer c] (.getId c)) :returns-fact true))
+
+(defrule vip-discount
+  [?customer <- min-id :from [Customer]]
+  =>
+  (apply insert! (let [fact-list ^List (ArrayList.)]
```

Comment:
this could be done with insert-all! (although I'm not sure that's really important here)